CSharp/Tests/Microsoft.Bot.Connector.Teams.Tests.Shared/TaskModuleTests.cs

```diff
@@ -0,0 +1,142 @@
+        }
+
+        /// <summary>
+        /// Test serialization of Task Module with Adaptive Card.
+        /// </summary>
+        [TestMethod]
+        public void TaskModuleTests_AdaptiveCard()
+        {
+            AdaptiveCards.AdaptiveCard currentAdaptiveCard = new AdaptiveCards.AdaptiveCard();
+            currentAdaptiveCard.Body.Add(new AdaptiveCards.AdaptiveTextBlock("some text on card"));
+
+            var action = new AdaptiveCards.AdaptiveOpenUrlAction();
+            action.Url = new Uri("https://microsoft.com");
+            currentAdaptiveCard.Actions.Add(action);
+
+            TaskModuleEnvelope currentTaskModuleEnvelope = new TaskModuleEnvelope
+            {
+                Task = new TaskModule
+                {
+                    Type = "continue",
+                    Value = new TaskModuleInfo()
+                    {
+                        Title = "Adaptive Card: Inputs",
+                        Card = currentAdaptiveCard.ToAttachment(),
+                        Height = "small",
+                        Width = "small",
+                    }
+                }
+            };
+
+            var expectedTaskModuleEnvelope = JsonConvert.DeserializeObject<TaskModuleEnvelope>(File.ReadAllText(@"Jsons\SampleTaskModuleAdaptiveCardPayload.json"));
+            Assert.IsTrue(expectedTaskModuleEnvelope != null);
+            Assert.IsTrue(expectedTaskModuleEnvelope.Task != null);
+            Assert.IsTrue(expectedTaskModuleEnvelope.Task.Type == currentTaskModuleEnvelope.Task.Type);
+            var expected = JsonConvert.DeserializeObject<TaskModuleInfo>(expectedTaskModuleEnvelope.Task.Value.ToString());
+            var current = currentTaskModuleEnvelope.Task.Value as TaskModuleInfo;
+            Assert.AreEqual(expected.Width, current.Width);
+            Assert.AreEqual(expected.Title, current.Title);
+            Assert.AreEqual(expected.Card.ContentType, current.Card.ContentType);
+        }
+
+        /// <summary>
+        /// Test serialization of Task Module with message payload.
+        /// </summary>
+        [TestMethod]
+        public void TaskModuleTests_Message()
+        {
+            TaskModuleEnvelope currentTaskModuleEnvelope = new TaskModuleEnvelope
+            {
+                Task = new TaskModule
+                {
+                    Type = "message",
+                    Value = "This is a test message"
+                }
+            };
+
+            var expectedTaskModuleEnvelope = JsonConvert.DeserializeObject<TaskModuleEnvelope>(File.ReadAllText(@"Jsons\SampleTaskModuleMessagePayload.json"));
+            Assert.IsTrue(expectedTaskModuleEnvelope != null);
+            Assert.IsTrue(expectedTaskModuleEnvelope.Task.Type == currentTaskModuleEnvelope.Task.Type);
+            Assert.IsTrue(expectedTaskModuleEnvelope.Task.Value.ToString() == currentTaskModuleEnvelope.Task.Value.ToString());
+
+        }
+    }
+}
```

Comment:
Please modify the TestBot to return a TaskModule response

Reply:
@siduppal, @robin-liao   - Therse are the Task Module response classes which are returned as payload in response like _Request.CreateResponse(HttpStatusCode.OK, taskEnvelope);_

Do you have any test cased which I could use as reference?